dashboard/app/controllers/lessons_controller.rb

```diff
@@ -11,14 +11,20 @@ class LessonsController < ApplicationController
   # them with the new lessons editor.
   def disallow_legacy_script_levels
     return unless @lesson.script_levels.reject(&:activity_section).any?
-    raise CanCan::AccessDenied.new(
-      "cannot edit lesson #{@lesson.id} because it contains legacy script levels"
-    )
+    return render :forbidden
   end
 
-  # GET /lessons/1
+  # GET /s/script-name/lessons/1
   def show
-    raise CanCan::AccessDenied.new("cannot view lesson #{@lesson.id} because it does not have a lesson plan on Code Studio") unless @lesson.has_lesson_plan && @lesson.script.is_migrated
+    script = Script.get_from_cache(params[:script_id])
+    raise ActiveRecord::RecordNotFound unless script
```

Comment:
I think this case is already covered here: https://github.com/code-dot-org/code-dot-org/blob/b1ce6a6ccd6cabfa906c8f3ac113365e58074b29/dashboard/app/models/script.rb#L490